### ISSUE TITLE
[Bug(ish)]: Poor edge scrolling on high refresh rate screens

### ISSUE THREAD

**Issue by wensimehrp** [OpenTTD/OpenTTD#11883]
### Version of OpenTTD

20240125 nightly

### Expected result

Smooth edge scrolling and arrow key movements.

### Actual result

Since it is difficult to describe the situation in plain text, here is a video explaining the problem.
Try playing it at 0.5x speed.

[2024-01-25 22-02-26.zip](https://github.com/OpenTTD/OpenTTD/files/14060827/2024-01-25.22-02-26.zip) (20240125)

[2024-01-25 22-11-23.zip](https://github.com/OpenTTD/OpenTTD/files/14060916/2024-01-25.22-11-23.zip) (20240123)

Basically, when using a display with a high refresh rate (in my example, 240hz, but the video is at 120hz), the quality of edge scrolling and arrow key movements is quite bad. I am not sure if this is how the game is supposed to work.

I've also tested on earlier versions (20240123) and it worked fine.

### Steps to reproduce

1. Start a game
2. Use edge scrolling or arrow keys to move the screen
